lib/broadway/processor.ex

```diff
@@ -168,17 +168,23 @@ defmodule Broadway.Processor do
     :telemetry.execute([:broadway, :processor, :message, :start], %{time: start_time}, metadata)
   end
 
-  defp emit_message_stop_event(start_time, metadata) do
+  defp emit_message_stop_event(start_time, metadata, handle_message_result) do
     stop_time = System.monotonic_time()
     measurements = %{time: stop_time, duration: stop_time - start_time}
+    updated_metadata = Map.put(metadata, :updated_message, handle_message_result)
```

Comment:
Sorry for the nitpick but instead of doing `Map.put`, let's build the whole map, it will be easier to see what is the actual metadata. We can build the map either outside or pass all parameters to build the map inside.

Reply:
No reason to apologize :). I genuinely appreciate the feedback and think that your suggestion makes it more explicit as to what you can expect in the metadata map. Fixed in latest commit!